comps/router/src/integrations/controllers/routellm_controller/routellm_controller.py

```diff
@@ -0,0 +1,61 @@
+import logging
+from comps.router.src.integrations.controllers.base_controller import BaseController
+from routellm.controller import Controller as RouteLLM_Controller
+# from decorators import log_latency
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+)
+
+class RouteLLMController(BaseController):
+    def __init__(self, config, hf_token=None, api_key=None, model_map=None):
+        self.config = config
+        # Use the provided model_map or default to an empty dict
+        self.model_map = model_map or {}
+        self.threshold = config.get("threshold", 0.2)
+        self.routing_algorithm = config.get("routing_algorithm")
+        self.controller_type = config.get("controller_type")
+
+        # Extract nested model info from model_map:
+        strong_model_info = self.model_map.get("strong", {})
+        weak_model_info = self.model_map.get("weak", {})
+
+        # Retrieve the model IDs from the nested structure
+        strong_model_name = strong_model_info.get("model_id")
+        weak_model_name = weak_model_info.get("model_id")
+
+        # Create the underlying RouteLLM_Controller with the unwrapped model names.
+        self.controller = RouteLLM_Controller(
+            routers=[self.routing_algorithm],
+            strong_model=strong_model_name,
+            weak_model=weak_model_name,
+            hf_token=hf_token,
+            api_key=api_key,
+            config=config.get("config"),
+        )
+
+    def route(self, messages):
```

Comment:
good to trace routing function with telemetry
```suggestion
    @opea_telemetry
    def route(self, messages):
```